src/keyszer/config_api.py

```diff
@@ -150,33 +150,41 @@ class TypingTooLong(Exception):
 class UnicodeNumberToolarge(Exception):
     pass
 
+unicode_combo_list = []
+
 
 def to_US_keystrokes(s):
     """
-    Turn alphanumeric string (with spaces and some ASCII) up to length of 100 characters into keystroke commands
+    Turn alphanumeric string (with spaces and some ASCII) up to length 
+    of 100 characters into keystroke commands
 
     Warn: Almost certainly not going to work with non-US keymaps.
     """
-    if len(s) > 100:
-        raise TypingTooLong("`to_keystrokes` only supports strings of 100 characters or less")
-
-    combo_list = []
-    for c in s:
-        if ord(c) > 127:
-            combos = unicode_keystrokes(ord(c))
-            combo_list.extend(combos)
-        elif c.isupper():
-            combo_list.append(combo("Shift-" + c))
-        elif (str.isalnum(c)):
-            combo_list.append(Key[c.upper()])
-        elif c in ASCII_TO_KEY:
-            combo_list.append(ASCII_TO_KEY[c])
-        elif c in ASCII_WITH_SHIFT:
-            combo_list.append(ASCII_WITH_SHIFT[c])
-        else:
-            raise CharacterNotSupported(f"The character {c} is not supported by `to_keystrokes` yet.")
-
-    return combo_list
+    def _to_keystrokes(ctx):
+        if len(s) > 100:
+            raise TypingTooLong("`to_keystrokes` only supports strings of 100 characters or less")
+        combo_list = []
+        for c in s:
+            if ord(c) > 127:
+                unicode_keystrokes(ord(c))(ctx)    # extra "()" updates global variable
+                combo_list.extend(unicode_combo_list)
```

Comment:
This needs to go - why can't the unicode_keystrokes command function (returned by the helper) be added directly to the combo_list itself?

Reply:
I didn't want `extend()` to get confused? 😆